src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/TemplatingPass.php

```diff
@@ -11,13 +11,20 @@
 
 namespace Symfony\Bundle\FrameworkBundle\DependencyInjection\Compiler;
 
+use Symfony\Bundle\FrameworkBundle\Templating\EngineInterface as FrameworkBundleEngineInterface;
 use Symfony\Component\DependencyInjection\ContainerBuilder;
 use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\Templating\EngineInterface;
 
 class TemplatingPass implements CompilerPassInterface
 {
     public function process(ContainerBuilder $container)
     {
+        if ($container->hasAlias('templating')) {
+            $definition = $container->findDefinition('templating');
+            $definition->setAutowiringTypes(array(EngineInterface::class, FrameworkBundleEngineInterface::class));
+        }
```

Comment:
This could be moved below the following `if` statement (there's no need to check if `templating` is an alias if it is a definition).